File: wjdrud2532_c++/programmers/12906.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    
    for(int i = 0; i < arr.size() - 1; i ++)
    {
        //===
        // if(i == arr.size() - 1)
        // {
        //     answer.push_back(arr[i]);
        //     break;
        // }
        //===
        
        if(arr[i] != arr[i + 1])
            answer.push_back(arr[i]);
    }
    answer.push_back(arr[arr.size() - 1]);


    return answer;
}

File: belif_python/programmers/12906.py
def solution(arr):
    result = [arr[0]]
    
    pre = arr[0]
    
    for num in arr:
        # 이전 것과 같다면 무시, 같지 않다면 추가
        if pre != num:
            result.append(num)
            pre = num
            
    return result